packages/ai-ide/src/browser/coder-agent.ts

```diff
@@ -30,6 +32,8 @@ export class CoderAgent extends AbstractStreamParsingChatAgent {
         identifier: 'openai/gpt-4o',
     }];
     protected defaultLanguageModelPurpose: string = 'chat';
+    readonly suggestions = [new MarkdownStringImpl(`Keep chats short and focused. [Start a new chat](command:${AI_CHAT_NEW_CHAT_WINDOW_COMMAND.id}) for a new task`
```

Comment:
Currently these are static. 
Question 1: To be able to also use this UI mechanism, we believe this mechanism should be dynamic. Do we want to allow the agent to push suggestions or do we pull from the chat, e.g. when a message is completed (Jonas thinks: push via the chat session)

Question 2: Currently, we use markdown String, which restricts us to commands. This might be weird, if we want to do something agent specific. Do we want to also allow real call backs, i.e. handing in functions here?

Reply:
Question 2.1: We should maybe add a button as well that opens something for showing even more actions, as there might be many suggetions, such as "update project-info" or common prompts. WDYT?

Reply:
Answers for now:
1.: Do it like the change set (push to the chat session)
2.: Go with an extended MarkdownString impl for now, tha also allows call backs to the agent. We should however ideally mke the rendering more flexible in the future, i.e. that you can register a render for a type of "SuggestionContent" to also allow other UI here in the future
2.1: I would introduce this in the default renderer, i.e. actions in the text plus optionally additional actions that are shown under "more" in a popup.

Reply:
Done apart from 2.1 - added as todo in description.

Reply:
I have created 2.1 as a follow up in https://github.com/eclipse-theia/theia/issues/15068 so we can close it here.